specification/under_development/Instruction_Set.md

```diff
@@ -1,36 +1,63 @@
+Instructions that execute on a classical processor or serve data transfer are
+expressed as LLVM instructions, including calls to `__quantum__rt__*` functions.
+These functions are forward declared in the IR and defined by the executing
+runtime. Which LLVM instructions and runtime functions are used/supported is
+captured in the QIR profile specification. The complete list of all runtime
+functions that are needed for executing QIR programs independent on which
+profile they have been compiled for is given in the [Library_Reference.md]().
+Most backends do not support the full set of runtime functions but merely a
+subset that is sufficient for the supported QIR profile(s). The [Base
+Profile](profiles/Base_Profile.md) defines the minimal set of runtime functions
+that need to be supported for by a backend to permit computations that make use
+of qubits and measurements of qubits.Additional functions may be needed to
+support the use of data types beyond those for qubit and measurement result
+values.
 
 ## Quantum Instruction Set (QIS)
 
-The QIR specification contains a [table](../QuantumInstructions.md) of commonly used instructions along with their signatures and a description of their functionality.
-Backends are **not** required to support all of these. Instead, each backend will declare which of these instructions it supports. We encourage to make instructions that are supported by a context independent implementation in terms of other instructions as a library rather than listing them as part of the backend specification. A library provided either in the form of a bitcode file can be linked in as part of a QIR compilation stage. 
+The QIR specification contains a [table](../QuantumInstructions.md) of commonly
+used instructions along with their signatures and a description of their
+functionality. Backends are **not** required to support all of these. Instead,
+each backend will declare which of these instructions it supports. We encourage
+to make instructions that are supported by a context independent implementation
+in terms of other instructions as a library rather than listing them as part of
+the backend specification. A library provided either in the form of a bitcode
```

Comment:
I wonder whether the use of "either" is grammatically correct here. Is there an alternative mentioned elsewhere?

Reply:
It is definitively not, and this sentence go scrambled/is incomplete; I intended to give a recommendation that is suitable for distribution, but haven't gotten around to actually confirm what a good one could be. I'll update this when I update the doc. This has also been called out by someone else, so I'll close the thread here and keep track of it above.